### PR TITLE
Japanese payment methods

### PR DESCRIPTION
**Pull request by mohammedDehairy**
Japanese payment methods:

- Convenience Stores
- Pay-easy ATM
- Online Banking
- 7-Eleven

### CODE REVIEW

File: Adyen/Utilities/PhoneExtensionsRepository.swift [hunk 1]
```diff
@@ -46,6 +49,8 @@ public enum PhoneNumberPaymentMethod {
                     "KR"]
         case .mbWay:
             return ["PT", "ES"]
+        case .generic:
+            return Array(allCountriesPhoneExtensions.keys).sorted()
```
Comment (descorp) [inline Adyen/Utilities/PhoneExtensionsRepository.swift:53]: No restrictions whatsoever?

Comment (mohammedDehairy) [reply Adyen/Utilities/PhoneExtensionsRepository.swift:53]: Yeah, I asked the designers, still waiting for an answer :-) .

File: Demo/Configuration.swift [hunk 2]
```diff
@@ -18,17 +18,17 @@ internal enum Configuration {
 
     static let appName = "Adyen Demo"
 
-    static let amount = Payment.Amount(value: 17408, currencyCode: "EUR")
+    static let amount = Payment.Amount(value: 17408, currencyCode: "JPY")
```
Comment (descorp) [inline Demo/Configuration.swift:21]: Let's uncommit that :D

**Review by descorp — DISMISSED** [review]

File: Demo/Configuration.swift (hunk 2 above)
Comment (mohammedDehairy) [reply Demo/Configuration.swift:21]: its fine, its not a secret, right? when the PR is done, I will revert it.

Comment (descorp) [reply Demo/Configuration.swift:21]: oh, it is WIP, sorry 😅

File: Adyen/Utilities/PhoneExtensionsRepository.swift (hunk 1 above)
Comment (mohammedDehairy) [reply Adyen/Utilities/PhoneExtensionsRepository.swift:53]: they said its okay for now

File: Demo/Common/Networking/PaymentsRequest.swift [hunk 3]
```diff
@@ -31,14 +31,17 @@ internal struct PaymentsRequest: Request {
         
         try container.encode(data.paymentMethod.encodable, forKey: .details)
         try container.encode(data.storePaymentMethod, forKey: .storePaymentMethod)
+        try container.encode(data.shopperName, forKey: .shopperName)
+        try container.encode(data.emailAddress ?? Configuration.shopperEmail, forKey: .shopperEmail)
+        try container.encode(data.telephoneNumber, forKey: .telephoneNumber)
+        try container.encode(Locale.current.identifier, forKey: .shopperLocale)
```
Comment (descorp) [inline Demo/Common/Networking/PaymentsRequest.swift:37]: Is this safe? What if shopper have dutch locale but wants english?
just wandering..

File: Adyen/Components/AbstractPersonalInformationComponent/FirstNameFormItemInjector.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+internal final class FirstNameFormItemInjector: FormItemInjector, Localizable {
```
Comment (descorp) [inline Adyen/Components/AbstractPersonalInformationComponent/FirstNameFormItemInjector.swift]: Both myself and Sonar think this is a duplication :D
We can probably address this in another PR..

File: Demo/Common/Networking/PaymentsRequest.swift (hunk 3 above)
Comment (mohammedDehairy) [reply Demo/Common/Networking/PaymentsRequest.swift:37]: well, merchants can have a settings page for the shopper to choose the locale, its totally up to merchants, right?

File: Adyen/Components/AbstractPersonalInformationComponent/FirstNameFormItemInjector.swift (hunk 4 above)
Comment (mohammedDehairy) [reply Adyen/Components/AbstractPersonalInformationComponent/FirstNameFormItemInjector.swift]: yep, this is fixed :-).

**Review by descorp — APPROVED** [review]

**Pull request merged**